libvast/vast/table_slice.hpp

```diff
@@ -103,11 +159,23 @@ class table_slice : public caf::ref_counted {
     return header_.rows;
   }
 
+  /// @returns a row view for the given `index`.
+  /// @pre `row < rows()`
+  row_view row(size_t index) const;
+
   /// @returns the number of rows in the slice.
   size_type columns() const noexcept {
     return header_.layout.fields.size();
   }
 
+  /// @returns a column view for the given `index`.
+  /// @pre `column < columns()`
+  column_view column(size_t index) const;
+
+  /// @returns a view for the column with given `name`.
+  /// @pre there is a c such that `c == column_name(name)`
+  column_view column(std::string_view name) const;
```

Comment:
I find the precondition hard to validate from the user perspective. It would require grabbing the layout and making sure one of the fields has the given name. Therefore, I'd change the interface so that `column` returns `optional<column_view>`. When the user is sure that the name exists, the deref is easy to perform. Otherwise the checking is explicit.

It also obviates the need to rely on exceptions for enforcing of preconditions. (For these, we usually use assertions anyway.)